ui/src/components/Banner.jsx

```diff
@@ -0,0 +1,272 @@
+import React, { useEffect } from 'react';
+
+import logo from './logo.png';
+import './Banner.scss';
+
+export default function Banner() {
+  useEffect(() => {
+    const image = main()
+    return () => { image.remove() }
+  }, [])
+
+  return <canvas className="Banner" id="glcanvas"></canvas>
+}
+
+function main() {
+  const image = new Image();
+  image.src = logo;
+  image.onload = function() { render(image); }
+  return image
+}
+
+function render(image) {
+  const canvas = document.getElementById('glcanvas');
+  canvas.setAttribute('width', image.width);
+  canvas.setAttribute('height', image.height);
+
+  if (window.location.href.startsWith('file:///')) {
+    alert(`You can't see the cool logo if you open this file directly from the filesystem :-(\n\nCheck docs/design/README.md for more information.`);
+    return;
+  }
+
+  const gl = canvas.getContext('webgl');
```

Comment:
you couldn't guess that, but we should also handle the case where gl is falsy (means webgl is not supported on that platform)